Assets/Scripts/Models/Buildable/Components/PowerConnection.cs

```diff
@@ -0,0 +1,208 @@
+        }
+        
+        public bool IsFull
+        {
+            get { return IsPowerAccumulator && AccumulatedPower.AreEqual(Provides.Capacity); }
+        }
+
+        public bool IsPowerProducer
+        {
+            get { return Provides != null && IsRunning && Provides.Rate > 0.0f && !IsPowerAccumulator; }
+        }
+
+        public bool IsPowerConsumer
+        {
+            get { return Requires != null && Requires.Rate > 0.0f && !IsPowerAccumulator; }
+        }
+
+        public bool IsPowerAccumulator
+        {
+            get { return Provides != null && Provides.Capacity > 0.0f; }
+        }
+
+        public override BuildableComponent Clone()
+        {
+            return new PowerConnection(this);
+        }
+
+        public override bool CanFunction()
+        {
+            bool hasPower = true;
+            if (IsPowerConsumer)
+            {
+                hasPower = World.Current.PowerNetwork.HasPower(this);
+            }
+
+            return hasPower;
+        }
+
+        public override void FixedFrequencyUpdate(float deltaTime)
+        {
+            bool areAllParamReqsFulfilled = true;
+            if (Requires != null)
+            {
+                areAllParamReqsFulfilled = AreParameterConditionsFulfilled(Requires.ParamConditions);
+            }
+
+            if (IsPowerAccumulator)
+            {
+                areAllParamReqsFulfilled &= AccumulatedPower > 0;
+            }
+
+            IsRunning = areAllParamReqsFulfilled;
+        }
+        
+        public override IEnumerable<string> GetDescription()
+        {           
+            string powerColor = IsRunning ? "lime" : "red";
+
+            yield return string.Format("Power Grid: <color={0}>{1}</color>", powerColor, IsRunning ? "Online" : "Offline");
+
+            if (IsPowerConsumer)
+            {
+                yield return string.Format("Power Input: <color={0}>{1}</color>", powerColor, Requires.Rate);
+            }
+
+            if (IsPowerProducer)
+            {
+                yield return string.Format("Power Output: <color={0}>{1}</color>", powerColor, Provides.Rate);
+            }
+
+            if (IsPowerAccumulator)
+            {
+                yield return string.Format("Power Accumulated: {0} / {1}", AccumulatedPower, Provides.Capacity);
+            }
+        }
+
+        protected override void Initialize()
+        {
+            // need to keep accumulator current charge in parameters
+            // TODO: need to prevent defining same type more than once
+            if (Provides != null)
+            {
+                if (Provides.Capacity > 0)
+                {
+                    FurnitureParams.AddParameter(new Parameter(CurAccumulatorChargeParamName, 0f));
+                    FurnitureParams.AddParameter(new Parameter(CurAccumulatorIndexParamName, 0));
+                }
+            }
+
+            FurnitureParams.AddParameter(new Parameter(CurIsRunningParamName, false));
+
```

Comment:
You don't need to explicitly AddParameter, due to the autocreation of Parameters when accessing a nonexistent one (in fact AddParameter just does `contents[parameter.GetName()] = parameter;`, it's an artifact from an older design and should have been removed), so `FurnitureParams[curAccumulatorChargeParamName] = 0f` would have exactly the same effect.